core/src/test/java/io/aiven/klaw/service/UsersTeamsControllerServiceTest.java

```diff
@@ -0,0 +1,186 @@
+    loginMock();
+  }
+
+  @Test
+  void getUserInfoDetails() {
+    String userId = "testuser";
+    when(handleDbRequests.getUsersInfo(anyString())).thenReturn(userInfo);
+    when(manageDatabase.getTeamNameFromTeamId(anyInt(), anyInt())).thenReturn("testteam");
+    UserInfoModel userInfoModel = usersTeamsControllerService.getUserInfoDetails(userId);
+    assertThat(userInfoModel.getTeamId()).isEqualTo(userInfo.getTeamId());
+    assertThat(userInfoModel.getUsername()).isEqualTo(userInfo.getUsername());
+    assertThat(userInfoModel.getRole()).isEqualTo(userInfo.getRole());
+    assertThat(userInfoModel.getTeam()).isEqualTo("testteam");
+  }
+
+  @Test
+  void updateProfile() throws KlawException {
+    UserInfoModel userInfoModel = utilMethods.getUserInfoMock();
+    when(handleDbRequests.getUsersInfo(anyString())).thenReturn(userInfo);
+    when(mailService.getUserName(any())).thenReturn("kwusera");
+    when(handleDbRequests.updateUser(any())).thenReturn(ApiResultStatus.SUCCESS.value);
+    ApiResponse apiResponse = usersTeamsControllerService.updateProfile(userInfoModel);
+    assertThat(apiResponse.getResult()).isEqualTo(ApiResultStatus.SUCCESS.value);
+  }
+
+  @Test
+  void updateProfileFailureDbUpdate() {
+    UserInfoModel userInfoModel = utilMethods.getUserInfoMock();
+    when(handleDbRequests.getUsersInfo(anyString())).thenReturn(userInfo);
+    when(mailService.getUserName(any())).thenReturn("kwusera");
+    when(handleDbRequests.updateUser(any())).thenThrow(new RuntimeException("Db update failed"));
+    KlawException thrown =
+        Assertions.assertThrows(
+            KlawException.class, () -> usersTeamsControllerService.updateProfile(userInfoModel));
+    assertThat(thrown.getMessage()).isEqualTo("Db update failed");
+  }
+
+  @Test
+  void updateUserNotAuthorized() throws KlawException {
+    UserInfoModel userInfoModel = utilMethods.getUserInfoMock();
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(true);
+    ApiResponse apiResponse = usersTeamsControllerService.updateUser(userInfoModel);
+    assertThat(apiResponse.getResult()).isEqualTo(ApiResultStatus.NOT_AUTHORIZED.value);
+  }
+
+  @Test
+  void updateUserNotAuthorizedToUpdateSuperAdmin() throws KlawException {
+    UserInfoModel userInfoModel = utilMethods.getUserInfoMock();
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(handleDbRequests.getUsersInfo(anyString())).thenReturn(userInfo);
+    when(commonUtilsService.getTenantId(anyString())).thenReturn(101);
+    when(mailService.getUserName(any())).thenReturn("testuser");
+    when(manageDatabase.getRolesPermissionsPerTenant(anyInt()))
+        .thenReturn(utilMethods.getRolesPermsMap());
+    ApiResponse apiResponse = usersTeamsControllerService.updateUser(userInfoModel);
+    assertThat(apiResponse.getResult())
+        .isEqualTo("Not Authorized to update another SUPERADMIN user.");
+  }
+
+  @Test
+  void getTeamDetails() {}
```

Comment:
Why do we have these empty methods?

Reply:
these tests are generated skeletons, need to be updated with actual tests